### DISCUSSION TITLE
How to disable specific rules in filter lists without to disable entire filter list?

### DISCUSSION THREAD

**Discussion by Dean-Corso** [uBlockOrigin/uAssets#21469] [Q&A] [answered]
Hi guys,

I have a question and looking for some info's & help how to disable single rules which get used on websites. I did check out the syntax list of uBO and found some disable filters like ghide / ehide / shide / badfilter I could use for cosmetic and elements filters but what about other filter rules like JavaScript & CSS styles? How to disable them without to disable entire filter list?

Example: JS & CSS style filter I found in Cookie Notices filter list.
```
website.com##+js(rc, modal-cacsp-open, , stay)
website.com##body,html:style(overflow: auto !important; position: initial !important;)
```
First filter is a scriptlet and second one a DOM. I would like to disable those two filters in my custom filter list and would like to use a other parameters for the CSS / DOM filter like this...

`website.com##body,html:style(overflow: visible !important; position: unset !important;)`
...so is this doable anyhow or not? Is there any way to blacklist specific filter rules which are listed any ANY filter list (not my own)? Problem is that I can not force my own custom rules in case of JS / CSS (DOM) etc. Just don't know it whether its possible or not yet.

Maybe you could post some tiny example rule commands about disabling filters / specific single filters etc. Thank you.

**Comment by mapx-** [comment]
```
website.com#@#+js(rc, modal-cacsp-open, , stay)
website.com#@#body,html:style(overflow: auto !important; position: initial !important;)
```

**Comment by Dean-Corso** [comment]
@mapx-,

thank you for that quick answer. Seems to work fine. 😃 Just didn't seen that info adding a @ sign between the ## ones. Good to know. I also see it now that I can disable those single filters in the network window clicking on the #@# button or in dom inspector view. My question in case of disabling filter by pressing the #@# button, where is the information is stored? Just wanna know it in case I want to undo it later again etc. Or is there any overview window of all changed rules (#@#) (crossed out) where I can edit / delete them etc?

**Reply by ghost** [reply]
This is not stored, it disappears after you close the logger window.

**Comment by mapx-** [comment]
All the info is here: https://github.com/gorhill/uBlock/wiki

see also:
https://help.adblockplus.org/hc/en-us/articles/360062733293

https://adguard.com/kb/general/ad-filtering/create-own-filters/

**Comment by Dean-Corso** [comment]
Thanks again for that info's guys.

Just have another extra question which is more off topic. Just would like to know how to work with all those rules I can set and how to analyze / debug any website to know what it does and to know how to prevent it using the right rules XY. I still found no tutorials / videos about that watching any pro people how they do it to see their approach steps. Only thing what I can do is checking out some filter rules from any lists and checking the websites to find out why they was set. No so easy to follow them. So one problem I always find on websites is that debugger detection to disallow using the dev tools of browser. Would be nice to find any source what could show detailed examples and explain them in detail to get more skilled in using entire uBO syntax correctly & effectively based on logically steps.

PS: Is there any uBO forum / board (not reddit) to ask questions etc?

**Reply by Yuki2718** [reply, marked as answer]
None of us have enough time or interest to create such tutorials, as we all are volunteers having real life and spending spare time for contribution. Those skills required are mostly the same as the ones required of any frontend developer, and there are plenty of existing materials for them including how to suppress devtools detection. Frankly, I want to say: oh, you found there's no good tutorials? That't nice, why don't you create yours by spending enough time to learn all of them before asking others to do so for free?
> Is there any uBO forum / board (not reddit) to ask questions etc?

https://www.reddit.com/r/uBlockOrigin/

**Reply by garry-ut99** [reply]
> https://www.reddit.com/r/uBlockOrigin/

The guy asked for: **not reddit**

**Reply by krystian3w** [reply]
hacker News (Y Combinator)?

**Comment by Dean-Corso** [comment]
@Yuki2718,

> None of us have enough time or interest to create such tutorials...

I think that's pretty pity. The uBO tool is great and it would be better when you guy's could bring people closer how to use it in the best way. I would like to use it for 100% if possible but I can't yet because of missing or too less information's / examples about specific filter types you can use. Otherwise people have to study by themself trying out filter rules XY you can find on any filter lists (I do too) and trying to back trace them to find out what they do and see the effects you know. Just did wonder because there are so many people using uBO but I found no advanced tutorials anywhere (only basic stuff) with the exception of Adguard documentation. Normally you can find tutorials (text or videos) for almost every known tools but not in case of uBO. Anyway, maybe just my opinion. Thanks.

**Reply by gorhill** [reply]
We don't have time to do what you are asking, it's not something you can argue.

**Reply by stephenhawk8054** [reply]
> as we all are volunteers having real life and spending spare time for contribution

Because of this.

---

You can investigate specific issues, and if there's something specific you don't understand or want to discuss, you can put them in details on uBO subreddit or github's discussions. That way it would be quicker and easier for others to help and also possible for you to gain more information/knowledge.

**Reply by Yuki2718** [reply]
> Otherwise people have to study by themself

All of us did this. Personally I'm happy to help those who continuously contribute, as it can be investment which in the long run returns back much more. But such people usually have done what he/she can before asking, and ask only a specific issue.

**Reply by Dean-Corso** [reply]
Hey guys,

thanks for your answers. Good OK, I will try to collect some issues I have and ask later about them here on discussions area when it's alright. Is it OK when I bundle my questions into one single thread etc to prevent creating new thread for any questions? Don't wanna blow up the discussions threads unnecessarily etc.

**Reply by stephenhawk8054** [reply]
You can put it in 1 thread. Before that, you can search if there were older threads discussing the same issue or not, then continue there.

**Reply by Yuki2718** [reply]
I hope you understand *answering question takes time* of volunteers. That time can be spent to actually improve the filters. Not saying do not ask, but do your own due diligence before asking.

**Comment by garry-ut99** [comment]
### 1
> Dean-Corso : I have a question and looking for some info's & help how to disable single rules which get used on websites. (...)
> Dean-Corso : thank you for that quick answer. Seems to work fine. 😃 Just didn't seen that info adding a `@` sign between the `##` ones. Good to know. 

What you're asking for, is already explained in the official documentation:
https://help.adblockplus.org/hc/en-us/articles/360062733293-How-to-write-filters#elemhide_exceptions

Also here partially : https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#scriptlet-injection
(mentioned how to disable all scriptlets on a site or on all sites, but the same method can be used to disable each filter separately)

<hr>

### 2
> Dean-Corso : I also see it now that I can disable those single filters in the network window clicking on the #@# button or in dom inspector view. My question in case of disabling filter by pressing the #@# button, where is the information is stored? Just wanna know it in case I want to undo it later again etc. Or is there any overview window of all changed rules (#@#) (crossed out) where I can edit / delete them etc?

Again, what you're asking for, is already explained in the official documentation :
https://github.com/uBlockOrigin/uBlock-issues/wiki/The-logger#temporary-exception-filters

<hr>

### 3
> Dean-Corso : Just have another extra question which is more off topic. Just would like to know how to work with all those rules I can set and how to analyze / debug any website to know what it does and to know how to prevent it using the right rules XY. (...)

Again, what you are asking for is already in the official documentation, all scriptlets are explained : 
https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library
(and all the other kind of filters are explained in the whole documentation)

And the other things like CSS/HTML/JS/websites it's out of scope for ad-blocker documentation,
you need to search internet for CSS/HTML/JS/websites coding documentation/tutorials.

<hr>

No offense, but no one likes people spamming tracker and asking questions which are already answered in the documentation, next time try to read / study the documentation more carefully before asking redundant questions, because we have too many people asking redundant questions and our time is limited, the purpose of documentation is to people not ask same questions over and over.
